packages/flutter_tools/test/general.shard/base/process_test.dart

```diff
@@ -211,52 +232,57 @@ void main() {
         flakyStdout.complete(<int>[]);
         return true;
       });
-      expect(() => processUtils.run(
+      expect(() => flakyProcessUtils.run(
         <String>['dummy'],
         timeout: delay - const Duration(milliseconds: 500),
         timeoutRetries: 0,
       ), throwsA(isInstanceOf<ProcessException>()));
-    }, overrides: <Type, Generator>{
-      ProcessManager: () => flakyProcessManager,
     });
   });
 
   group('runSync', () {
     ProcessManager mockProcessManager;
+    ProcessUtils processUtils;
+    BufferLogger testLogger;
 
     setUp(() {
       mockProcessManager = MockProcessManager();
+      testLogger = BufferLogger(
```

Comment:
There is probably some room to create a `TestLogger` utility here for BufferLogger specifically. This does seem a bit boilerplaty

Reply:
Yeah, I expect that as we move up the hierarchy, this will show up a lot more.